marlowe-runtime-web/server/Language/Marlowe/Runtime/Web/Server/REST/ApiError.hs

```diff
@@ -111,43 +110,32 @@ rangeNotSatisfiable msg errorCode = toServerError . ApiError msg (fromMaybe "Ran
 rangeNotSatisfiable' :: String -> ServerError
 rangeNotSatisfiable' msg = rangeNotSatisfiable msg Nothing
 
-instance HasDTO (WithdrawError 'V1) where
-  type DTO (WithdrawError 'V1) = ApiError
+instance HasDTO WithdrawError where
+  type DTO WithdrawError = ApiError
 
-instance ToDTO (WithdrawError 'V1) where
+instance ToDTO WithdrawError where
   toDTO = \case
     WithdrawEraUnsupported era -> ApiError ("Current network era not supported: " <> show era) "WithdrawEraUnsupported" Null 503
-    WithdrawConstraintError (MintingUtxoNotFound _) -> ApiError "Minting UTxO not found" "MintingUtxoNotFound" Null 500
-    WithdrawConstraintError (RoleTokenNotFound _) -> ApiError "Role token not found" "RoleTokenNotFound" Null 403
-    WithdrawConstraintError ToCardanoError -> ApiError "Internal error" "ToCardanoError" Null 500
-    WithdrawConstraintError MissingMarloweInput -> ApiError "Internal error" "MissingMarloweInput" Null 500
-    WithdrawConstraintError (PayoutInputNotFound _) -> ApiError "Internal error" "PayoutInputNotFound" Null 500
-    WithdrawConstraintError (CalculateMinUtxoFailed _) -> ApiError "Internal error" "CalculateMinUtxoFailed" Null 500
-    WithdrawConstraintError (CoinSelectionFailed msg) -> ApiError ("Coin selection failed: " <> msg) "CoinSelectionFailed" Null 400
-    WithdrawConstraintError (BalancingError _) -> ApiError "Internal error" "BalancingError" Null 500
-    WithdrawLoadMarloweContextFailed LoadMarloweContextErrorNotFound -> ApiError "Marlowe contract not found" "MarloweContractNotFound" Null 404
-    WithdrawLoadMarloweContextFailed (LoadMarloweContextErrorVersionMismatch _) -> ApiError "Marlowe contract version mismatch" "MarloweContractVersionMismatch" Null 400
-    WithdrawLoadMarloweContextFailed LoadMarloweContextToCardanoError -> ApiError "Internal error" "LoadMarloweContextToCardanoError" Null 500
-    WithdrawLoadMarloweContextFailed (MarloweScriptNotPublished _) -> ApiError "Internal error" "MarloweScriptNotPublished" Null 500
-    WithdrawLoadMarloweContextFailed (PayoutScriptNotPublished _) -> ApiError "Internal error" "PayoutScriptNotPublished" Null 500
-    WithdrawLoadMarloweContextFailed (ExtractCreationError _) -> ApiError "Internal error" "ExtractCreationError" Null 500
-    WithdrawLoadMarloweContextFailed (ExtractMarloweTransactionError _) -> ApiError "Internal error" "ExtractMarloweTransactionError" Null 500
-    UnableToFindPayoutForAGivenRole _ -> ApiError "No payouts available for given role" "UnableToFindPayoutForAGivenRole" Null 409
-
-instance HasDTO (CreateError 'V1) where
-  type DTO (CreateError 'V1) = ApiError
-
-instance ToDTO (CreateError 'V1) where
+    WithdrawConstraintError err -> ApiError (show err) "ConstraintError" Null case err of
+      MintingUtxoNotFound _ -> 400
+      RoleTokenNotFound _ -> 400
+      PayoutNotFound _ -> 400
+      CoinSelectionFailed _ -> 400
+      _ -> 500
```

Comment:
Doesn't this change make error reporting *less* expressive? The error reports were already too terse.

Reply:
It doesn't make it less expressive, it actually makes it more expressive:

```hs
-- before on "PayoutInputNotFound"
ApiError "Internal error" "PayoutInputNotFound" Null 500

-- after
ApiError "PayoutNotFound <payout tx in>" "ConstraintError" 500